src/renderers/noop/ReactNoop.js

```diff
@@ -146,6 +149,10 @@ var ReactNoop = {
     ReactNoop.flushLowPri();
   },
 
+  performHighPriWork(fn: Function) {
+    NoopRenderer.performWithPriority(HighPriority, fn);
```

Comment:
Yea, I think the deprioritization case needs to be handled by the renderer somehow. However, I don't see a good way to expose it as an API to schedule a specific priority.

I think possibly that `prepareUpdate` could return the priority level. However, currently that happens after the children has already rendered. Would need to split children update from props update - which I want to do anyway.

It could also be an explicit API like `isOffscreen(...)` or something.

I'm not sure if the deprioritization use case make much sense for a renderer to do in another case than `Offscreen` so it probably isn't granular anyway.